### PR TITLE
take  Set as track of fragmentNames which is already visited.

### PR DESCRIPTION
**Pull request by dugenkui03**
In `FieldCollector`, take `Set` as track of fragmentNames which is already visited.

### CODE REVIEW

File: src/main/java/graphql/execution/FieldCollector.java [hunk 1]
```diff
@@ -32,7 +33,7 @@
 
     public MergedSelectionSet collectFields(FieldCollectorParameters parameters, MergedField mergedField) {
         Map<String, MergedField> subFields = new LinkedHashMap<>();
-        List<String> visitedFragments = new ArrayList<>();
+        Set<String> visitedFragments = new HashSet<>();
```
Comment (andimarek) [inline src/main/java/graphql/execution/FieldCollector.java:36]: We should use `LinkedHashSet` here to be predictable and make debugging easier.

File: src/main/java/graphql/execution/FieldCollector.java [hunk 2]
```diff
@@ -52,13 +53,13 @@ public MergedSelectionSet collectFields(FieldCollectorParameters parameters, Mer
      */
     public MergedSelectionSet collectFields(FieldCollectorParameters parameters, SelectionSet selectionSet) {
         Map<String, MergedField> subFields = new LinkedHashMap<>();
-        List<String> visitedFragments = new ArrayList<>();
+        Set<String> visitedFragments = new HashSet<>();
```
Comment (andimarek) [inline src/main/java/graphql/execution/FieldCollector.java:56]: same as above: We should use LinkedHashSet here to be predictable and make debugging easier.

File: src/main/java/graphql/execution/FieldCollector.java (hunk 1 above)
Comment (dugenkui03) [reply src/main/java/graphql/execution/FieldCollector.java]: done

**Comment by andimarek** [comment]
thanks

**Pull request merged**